libbeat/metric/system/process/process.go

```diff
@@ -359,18 +380,57 @@ func GetProcCPUPercentage(s0, s1 *Process) (normalizedPct, pct, totalPct float64
 	if s0 != nil && s1 != nil {
 		timeDelta := s1.SampleTime.Sub(s0.SampleTime)
 		timeDeltaMillis := timeDelta / time.Millisecond
-		totalCPUDeltaMillis := int64(s1.Cpu.Total - s0.Cpu.Total)
+		totalCPUDeltaMillis := int64(s1.CPU.Total - s0.CPU.Total)
 
 		pct := float64(totalCPUDeltaMillis) / float64(timeDeltaMillis)
 		normalizedPct := pct / float64(runtime.NumCPU())
-
 		return common.Round(normalizedPct, common.DefaultDecimalPlacesCount),
 			common.Round(pct, common.DefaultDecimalPlacesCount),
-			common.Round(float64(s1.Cpu.Total), common.DefaultDecimalPlacesCount)
+			common.Round(float64(s1.CPU.Total), common.DefaultDecimalPlacesCount)
 	}
 	return 0, 0, 0
 }
 
+// GetCgroupPercentage returns CPU usage percentages for a given cgroup
+// see GetProcCPUPercentage for implementation details, as the two are conceptually similar.
+// Note that the cgroup controller reports system and user times in USER_HZ, while
+// totals are reported in nanoseconds. Because of this, any math that mixes the two might be slightly off,
+// as USER_HZ is less precise value that will get rounded up to nanseconds.
+// Because of that, `user` and `system` metrics reflect a precentage of overall CPU time, but can't be compared to the total pct values.
+func GetCgroupPercentage(s0, s1 *Process) (pct, normalizedPct, userPct, systemPct, normalizedUser, normalizedSystem float64) {
+	if s0 != nil && s1 != nil && s0.RawStats != nil && s1.RawStats != nil && s0.RawStats.CPUAccounting != nil && s1.RawStats.CPUAccounting != nil {
```

Comment:
Nit. Consider reversing this condition to save a nesting level and return earlier in case of lacking the expected info.
```suggestion
	if s0 == nil || s1 == nil || s0.RawStats == nil || s1.RawStats == nil || s0.RawStats.CPUAccounting == nil || s1.RawStats.CPUAccounting == nil {
		return 0, 0, 0, 0, 0, 0
	}
```